### PR TITLE
bpo-37609: Add device path support in ntpath splitdrive

### PR DESCRIPTION
**Pull request by nsiregar**
Add device path support in `ntpath.splitdrive`

https://bugs.python.org/issue37609

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2019-07-18-14-39-59.bpo-37609.1ZYNbG.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add support for unc device path in :mod:`ntpath.splitdrive`
```
Comment (ZackerySpytz) [inline Misc/NEWS.d/next/Library/2019-07-18-14-39-59.bpo-37609.1ZYNbG.rst]: Using the mod role here is not correct. Please see https://devguide.python.org/documenting/#id4.

Comment (nsiregar) [reply Misc/NEWS.d/next/Library/2019-07-18-14-39-59.bpo-37609.1ZYNbG.rst]: Thanks, I've updated it to use `meth`.

Comment (ZackerySpytz) [reply Misc/NEWS.d/next/Library/2019-07-18-14-39-59.bpo-37609.1ZYNbG.rst]: It should be `func`.

Comment (nsiregar) [reply Misc/NEWS.d/next/Library/2019-07-18-14-39-59.bpo-37609.1ZYNbG.rst]: Thanks, I've updated it

**Comment by zooba** [comment]
Sorry for not seeing this earlier, I have a few general suggestions:
* put a leading underscore on the new functions (or else add full documentation for them)
* use constant strings in the comparisons, rather than `sep*2`
* add some tests using backslashes as well, not just forward slashes

File: Lib/ntpath.py [hunk 2]
```diff
@@ -159,11 +157,30 @@ def splitdrive(p):
             if index2 == -1:
                 index2 = len(p)
             return p[:index2], p[index2:]
-        if normp[1:2] == colon:
+        if is_drive_path(normp, colon):
             return p[:2], p[2:]
     return p[:0], p
 
 
+def is_unc_path(path, sep):
```
Comment (eryksun) [inline Lib/ntpath.py:165]: These functions are clearly private (e.g. parameterized by `sep` and `colon`), so the function names should have a leading underscore.

**Comment by nsiregar** [comment]
working on it

**Comment by eryksun** [comment]
For an alternate and more complete implementation, see the splitdrive.py file that I attached to bpo-37609.

**Comment by barneygale** [comment]
I managed to fix the issue elsewhere (in #91882), so I'm going to close this PR, but thank you very much for taking a look at it! :)

**Pull request closed without merging**